sdk/apimanagement/Azure.ResourceManager.ApiManagement/src/autorest.md

```diff
@@ -304,6 +313,8 @@ directive:
       $.PortalSettingsContractProperties.properties.subscriptions['x-ms-client-name'] = 'IsSubscriptions';
       $.PortalSettingsContractProperties.properties.userRegistration['x-ms-client-name'] = 'IsUserRegistration';
       $.PrivateEndpointConnectionRequest.properties.id['x-ms-format'] = 'arm-id';
+      delete $.ContentTypeContract.allOf;
```

Comment:
Because the id of content type and content item is not resource identifier, therefore here we delete the `allof` to prevent the model from inheriting `ResourceData`.

Reply:
Therefore as a result, we lose the `type` property on this model. Is this expected or a workaround?

Reply:
I believe this is just a workaround. I didn't find a way to prevent this model from inheriting `ResourceData`, therefore I decided to delete it. Once we have a better way to support this case, we can add it back as there is no breaking change.

Reply:
Could you please open an issue in atuorest.csharp? We might need a new configuration, or extend some existing configuration to support this.

Reply:
Opened here: https://github.com/Azure/autorest.csharp/issues/2639